configuration/debian/tedge_agent/preinst

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+set -e
+
+purge_agent_directory() {
+    if [ -d "/etc/tedge/.agent" ]; then
+        rm -rf /etc/tedge/.agent
+    fi
+}
+
+purge_agent_lock() {
+   if [ -f "/run/lock/tedge_agent.lock" ]; then
+       rm -rf /run/lock/tedge_agent.lock
+   fi
+}
+
+purge_var_data_directory() {
+    if [ -d "/var/tedge" ]; then
+        rm -rf /var/tedge
+    fi
+}
+
+echo "Remove stale runtime directories"
+purge_agent_directory
+purge_agent_lock
+purge_var_data_directory
+
+#DEBHELPER#
```

Comment:
Why is there an issue if the directories and file are already there? Seems like `sudo apt remove --purge` is losing meaning if we purge anyway

Reply:
In my opinion, these are runtime directories which shouldn't be carried forward from one release to another. The "purge" vs "remove" logic should only be applied for "configuration files" and things like that, which can be migrated from one version to another. It doesn't make much sense to migrate a runtime directory.

But, with that logic, I'd agree that the `postrm` script should also be updated to delete these directories on `remove` as well as `purge` so that it is consistent with this `preinst` script.

Reply:
> Why is there an issue if the directories and file are already there? Seems like `sudo apt remove --purge` is losing meaning if we purge anyway

To answer the question, "what the issue was": If one version creates these directories as `root` and the next version expects these directories to be owned by `tedge`, this script will help in dealing with such issues.

Reply:
We need to be really cautious here.

Do not forget that `tedge_agent` might be used to update itself. I.e a new version of the agent can be installed by the apt-plugin while the previous version of the agent is still running and driving the apt-plugin. The former must not kill the latter.

@rina23q's experiment on updating `tedge_agent` from the cloud using the apt-plugin demonstrated that the issues are related on these `postrm` and `preinst` scripts. See https://github.com/thin-edge/thin-edge.io/issues/1466.

Reply:
Reverted the addition of this `preinst` script.